health/health.go

```diff
@@ -0,0 +1,93 @@
+package health
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	health "github.com/ONSdigital/dp-healthcheck/healthcheck"
+	rchttp "github.com/ONSdigital/dp-rchttp"
+	"github.com/ONSdigital/log.go/log"
+)
+
+// ErrInvalidAPIResponse is returned when an api does not respond
+// with a valid status
+type ErrInvalidAPIResponse struct {
+	expectedCode int
+	actualCode   int
+	uri          string
+}
+
+// Client represents an api client
+type Client struct {
+	client rchttp.Clienter
+	name   string
+	url    string
+}
+
+// New creates a new instance of Client with a given api url
+func New(name, url string) *Client {
+	c := &Client{
+		client: rchttp.NewClient(),
+		name:   name,
+		url:    url,
+	}
+
+	// Set the number of max retries to 1, allow caller to
+	// handle re-calling health checker
+	c.client.SetMaxRetries(1)
```

Comment:
I wonder if accepting the max retries as a param to the function might be better? The logic here makes sense but it's assuming that this is always the case. By passing it as a parameter you are delegating that decision to the caller.